File: LV 2/k진수에서 소수의 개수 구하기.py
import math
import re

def convert(n, k):
    cv = ""
    while n != 0:
        cv = str(n % k) + cv
        n //= k
    
    return cv

def check_prime(n):
    if n == 1:
        return False
    
    for i in range(2, int(math.sqrt(n)) + 1):
        if n % i == 0:
            return False
        
    return True

def solution(n, k):
    PATTERN = r"0+"
    answer = 0
    convert_string = convert(n, k)
    st_list = re.split(PATTERN, convert_string)
    
    for st in st_list:
        if st == "":
            continue
        if check_prime(int(st)):
            answer += 1
        
    return answer

File: LV 2/k진수에서 소수의 개수 구하기.java
import java.util.Arrays;

class Solution {
    public String toKDecimal(int n, int k) {
        StringBuilder builder = new StringBuilder();
        if (n == 0) {
            return "";
        }
        while (n > 0) {
            builder.append(n % k);
            n = n / k;
        }
        builder.reverse();
        return builder.toString();
    }

    public boolean isPrime(final Long n) {
        if (n < 2) {
            return false;
        }

        if (n == 2) {
            return true;
        }

        for (long i = 2; i < (int)Math.sqrt(n) + 1; i++) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }

    public int solution(int n, int k) {
        int answer = 0;
        String convertKDecimal = toKDecimal(n, k);
        String[] elements = convertKDecimal.split("0+");
        for (String element : elements) {
            if (isPrime(Long.parseLong(element))) {
                answer += 1;
            }
        }
        return answer;
    }

}
